Source/Scene/Cesium3DTileStyleEngine.js

```diff
@@ -74,36 +76,41 @@ define([
                 //   2) this tile is now visible, but it wasn't visible when the style was first assigned
                 if (tile.lastStyleTime !== lastStyleTime) {
                     tile.lastStyleTime = lastStyleTime;
-                    styleCompositeContent(this, tile.content, stats);
+                    styleCompositeContent(this, frameState, tile.content, stats);
 
                     ++stats.numberOfTilesStyled;
                 }
             }
         }
     };
 
-    function styleCompositeContent(styleEngine, content, stats) {
+    function styleCompositeContent(styleEngine, frameState, content, stats) {
         var innerContents = content.innerContents;
         if (defined(innerContents)) {
             var length = innerContents.length;
             for (var i = 0; i < length; ++i) {
                 // Recurse for composites of composites
-                styleCompositeContent(styleEngine, innerContents[i], stats);
+                styleCompositeContent(styleEngine, frameState, innerContents[i], stats);
             }
         } else {
             // Not a composite tile
-            styleContent(styleEngine, content, stats);
+            styleContent(styleEngine, frameState, content, stats);
         }
     }
 
     var scratchColor = new Color();
 
-    function styleContent(styleEngine, content, stats) {
+    function styleContent(styleEngine, frameState, content, stats) {
         var length = content.featuresLength;
         var style = styleEngine._style;
 
         stats.numberOfFeaturesStyled += length;
 
+        // Apply style to point cloud. Only apply style if the point cloud is not backed by a batch table.
+        if ((content instanceof PointCloud3DTileContent) && (length === 0)) {
+            content.applyStyle(frameState, style);
```

Comment:
Perhaps rework this function so that each content type has a function like `applyStyleWithShader` that returns a boolean.  If it returns `false` then the style was not applied and the loop below can be used.  This avoids the special cases and reduces coupling.